geode-core/src/main/java/org/apache/geode/distributed/internal/InternalLocator.java

```diff
@@ -1155,42 +1155,43 @@ private void restartWithSystem(InternalDistributedSystem newSystem, InternalCach
         throw new IllegalStateException(
             "A locator can not be created because one already exists in this JVM.");
       }
-      internalDistributedSystem = newSystem;
-      internalCache = newCache;
-      internalDistributedSystem.setDependentLocator(this);
-      logger.info("Locator restart: initializing TcpServer");
-
-      try {
-        server.restarting(newSystem, newCache, configurationPersistenceService);
-      } catch (CancelException e) {
-        internalDistributedSystem = null;
-        internalCache = null;
-        logger.info("Locator restart: attempt to restart location services failed", e);
-        throw e;
-      }
+    }
+    internalDistributedSystem = newSystem;
+    internalCache = newCache;
+    internalDistributedSystem.setDependentLocator(this);
+    logger.info("Locator restart: initializing TcpServer");
 
-      if (productUseLog.isClosed()) {
-        productUseLog.reopen();
-      }
+    try {
+      server.restarting(newSystem, newCache, configurationPersistenceService);
+    } catch (CancelException e) {
+      internalDistributedSystem = null;
+      internalCache = null;
+      logger.info("Locator restart: attempt to restart location services failed", e);
+      throw e;
+    }
 
-      productUseLog.monitorUse(newSystem);
+    if (productUseLog.isClosed()) {
+      productUseLog.reopen();
+    }
 
-      if (isSharedConfigurationEnabled()) {
-        configurationPersistenceService =
-            new InternalConfigurationPersistenceService(newCache);
-        startClusterManagementService();
-      }
+    productUseLog.monitorUse(newSystem);
 
-      if (!server.isAlive()) {
-        logger.info("Locator restart: starting TcpServer");
-        startTcpServer();
-      }
+    if (isSharedConfigurationEnabled()) {
```

Comment:
This looks like it will fix the problem.  I wish the lifecycle of the locator was less complicated.